File: backend/src/main/java/com/github/flexca/sbae/backend/users/model/UserSearchRequest.java
package com.github.flexca.sbae.backend.users.model;

import com.github.flexca.sbae.backend.common.model.search.OrderDirection;
import lombok.Data;

@Data
public class UserSearchRequest {

    private String organizationId;
    private String id;
    private String email;
    private String name;
    private String createdFrom;
    private String createdTo;
    private String status;

    private int limit = 20;
    private int offset = 0;

    private String orderBy;
    private OrderDirection orderDirection = OrderDirection.DESC;
}

File: frontend/src/app/organization/user-search.request.ts
export class UserSearchRequest {

    constructor(
        public organizationId: string,
        public limit: number,
        public offset: number,
        public id?: string,
        public email?: string,
        public name?: string,
        public createdFrom?: string,
        public createdTo?: string,
        public status?: string
        ) {  

    }

}